pkg/distributor/distributorerror/error.go

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributorerror
+
+import (
+	"fmt"
+
+	"github.com/grafana/mimir/pkg/util/globalerror"
+	"github.com/grafana/mimir/pkg/util/validation"
+)
+
+// DistributorPushError is a generic error returned on
+// distributor's write path.
+type DistributorPushError struct {
+	err error
+}
```

Comment:
Why do we still need this?

Reply:
@colega 
Similarly as I explained above, my idea was to have all the errors returned by the distributor be of the same type.
This wrapper is now used for the following errors:
- `haTracker.updateKVStore()`
- `errMaxInflightRequestsReached`
- `errMaxIngestionRateReached`
- `errMaxInflightRequestsBytesReached`
They will all be treated as an internal server error. This wrapper would not add any additional benefits. 
But, since the last 3 are all instance-limit errors, maybe it wouldn't be bad to add another error type, `InstanceLimitError`, to group the last 3 errors together, and to remove `DistributorPushError`. WDYT?

Reply:
> But, since the last 3 are all instance-limit errors, maybe it wouldn't be bad to add another error type, InstanceLimitError

Would they be treated in a special way or would they still be an internal server error?

Reply:
Currently they would be translated int an internal server error.

Reply:
I have removed the `NewDistributorPushError()` constructor and its usages, and I renamed `DistributorPushError` into `distributorPushError` and used it locally in `distributorerror/error.go` only.

Reply:
Then I would not add an extra type, as it would not help.